diesel_derives/src/insertable.rs

```diff
@@ -26,27 +26,11 @@ pub fn derive(item: syn::DeriveInput) -> Result<proc_macro2::TokenStream, Diagno
     impl_generics.params.push(parse_quote!('insert));
     let (impl_generics, ..) = impl_generics.split_for_impl();
 
-    let (direct_field_ty, direct_field_assign): (Vec<_>, Vec<_>) = model
-        .fields()
-        .iter()
-        .map(|f| {
-            (
-                (field_ty(f, table_name, None)),
-                (field_expr(f, table_name, None)),
-            )
-        })
-        .unzip();
-
-    let (ref_field_ty, ref_field_assign): (Vec<_>, Vec<_>) = model
-        .fields()
-        .iter()
-        .map(|f| {
-            (
-                (field_ty(f, table_name, Some(quote!(&'insert)))),
-                (field_expr(f, table_name, Some(quote!(&)))),
-            )
-        })
-        .unzip();
+    let direct_field_ty = model.fields().iter().map(|f| field_ty(f, table_name, None)).collect::<Result<Vec<_>, _>>()?;
+    let direct_field_assign = model.fields().iter().map(|f| field_expr(f, table_name, None)).collect::<Result<Vec<_>, _>>()?;
+
+    let ref_field_ty = model.fields().iter().map(|f| field_ty(f, table_name, Some(quote!(&'insert)))).collect::<Result<Vec<_>, _>>()?;
+    let ref_field_assign = model.fields().iter().map(|f| field_expr(f, table_name, Some(quote!(&)))).collect::<Result<Vec<_>, _>>()?;
```

Comment:
To my knowledge, `unzip` doesn't support the `Vec<Result>` -> `Result<Vec>` conversion. I didn't actually try though to be honest because I wanted to leave optimizations like this up to the end when there is a general ACK on the concept.

Reply:
Good point about `unzip()` doesn't support that. So maybe just go back to a normal loop here?